pkg/deploy/deploy.go

```diff
@@ -127,7 +141,12 @@ func execute(environment environment.Environment, projects []project.Project, dr
 			}
 
 			if err != nil {
-				return err
+				// by default stop deployment on error
+				if !continueOnError && !dryRun {
```

Comment:
Small nitpick: It would be cool if we could turn this condition to an `||` 
without negations. This would be easier to read and grasp imho .